rfcs/convergence/dependency-versions.md

```diff
@@ -0,0 +1,91 @@
+# RFC: Use ranges to define versions of dependencies
+
+@layershifter @ling1726 @miroslavstastny
+
+## Summary
+
+This RFC proposes to continue use caret (`^`) and tilde (`~`) for production dependencies of converged packages.
+
+## Problem statement
+
+The are two options to manage packages:
+
+- ranges via carets (`^1.0.0`), tildes (`~1.0.0`) or signs (`>=1.0.0`, `1.0.0 || 2.0.0`)
+- exact versions
+
+The preferred approach may vary on use case: applications and libraries follow different practices. Fluent UI is a library, so we should focus on the best experience for customers.
+
+Usage of exact versions in our dependencies solves a problem with different versions for us and customers so there will be a guarantee that they will use same versions as we. This should guarantee us there will be no unknown bugs caused by third party dependencies in different versions. This is a huge benefit.
+
+However, an approach with exact versions creates a different set of problems for consumers. Let's assume that we have two packages and ours will use an exact version:
+
+```json
+{
+  "name": "@fluentui/react-COMPONENT",
+  "dependencies": {
+    "lodash": "4.17.20"
+  }
+}
+```
+
+```json
+{
+  "name": "eslint",
+  "dependencies": {
+    "lodash": "^4.17.21"
+  }
+}
+```
+
+This will result in two versions of installed packages by a package manager (`yarn` in this case):
+
+```
+lodash@4.17.20:
+  version "4.17.20"
+  resolved "https://..."
+  integrity sha512-...
+
+lodash@^4.17.21:
+  version "4.17.21"
+  resolved "https://..."
+  integrity sha512-...
+```
+
+Thus a single package gets included multiple times in a Webpack bundle due to different package versions. This situation will happen without any warning, resulting in extra bloat in consumer's bundle and may lead to hard-to-find bugs.
+
+It's also harder for consumers in this case to consume security patches, for example `lodash@4.17.20` contains [a security vulnerability](https://snyk.io/vuln/SNYK-JS-LODASH-1018905) but this version will be inside consumer's application without additional movements from their side. If we will use a caret ("lodash": "^4.17.20") only a single version (a latest in a range) will be installed:
```

Comment:
> If we will use a caret ("lodash": "^4.17.20") only a single version (a latest in a range) will be installed

Just a heads up, this is not necessarily true--if a dep on `lodash@^4.17.20` resolved to `4.17.20` already exists in the lockfile, and that semver spec is still referenced by another package, yarn may add a separate entry for `lodash@^4.17.21` => `4.17.21` instead of resolving both to `4.17.21`. 

(Doesn't affect the proposal since this is an issue with yarn in general, but I wanted to call it out for awareness.)

Reply:
Yeah, you're totally right 👍 But with ranges we can at least deduplicate dependencies manually or via tools ([`yarn-deduplicate` for Yarn v1](https://github.com/atlassian/yarn-deduplicate) or [`yarn dedupe` for V2](https://github.com/yarnpkg/berry/pull/1558)).